### ISSUE TITLE
Trying to build against headers/libs coming from a Nuget package in CMake

### ISSUE THREAD

**Issue by MSLukeWest** [dotnet/arcade#6394]
[ x] This issue is causing unreasonable pain

I'm working in the dotnet deployment-tools repo that uses the Arcade toolset.  I'm working on a change that uses headers and libs from this WiX nuget package: https://www.nuget.org/packages/WiX/3.11.2.  Am I supposed to just copy the necessary headers and libs into the repo or is there a more elegant way to reference the Nuget package?  If so, can you point me to an example?  I've looked for examples in some of the dotnet repos but haven't found anything.

**Comment by markwilkie** [comment]
Any chance you can help out @alexperovich ?

**Comment by alexperovich** [comment]
I believe this (https://stackoverflow.com/a/56093754) would be the correct way to do this. As of version 3.15 cmake has built-in support for emitting package references into the vcxproj.

**Comment by MSLukeWest** [comment]
@alexperovich  Thanks for the reply.  This gets me a PackageReference element in the vcxproj, but it's not used in any way.  Any idea how to make the package actually get downloaded so it can be used by the build?

**Comment by alexperovich** [comment]
The discussion here https://developercommunity.visualstudio.com/idea/351636/use-packagereference-in-vcxproj.html suggests that just sticking the property `<_NuGetTargetFallbackMoniker>$(_NuGetTargetFallbackMoniker);native,Version=v0.0</_NuGetTargetFallbackMoniker>` in a Directory.Build.props that the vcxprojs share will make `msbuild <project> /t:Restore` work, because the nuget stuff is already there. I don't know what else would be needed to wire up the headers and libs, but a bit more msbuild goop could do it.

**Comment by MSLukeWest** [comment]
Thanks @alexperovich for the replies and ideas.  I still wasn't exactly sure how to hook this all up and eventually switched strategies after realizing that what I needed from these libraries could be very easily re-implemented in my code, so I went that route.

Feel free to close this issue, or leave it open until there's a working example.

**Comment by alexperovich** [comment]
TLDR: cmake can emit PackageReference items into the vcxproj files. It is up to the package author and/or consumer to wire up libs and headers.
